source/Application/Controller/Admin/LanguageMain.php

```diff
@@ -507,11 +501,7 @@ protected function _validateInput()
      */
     protected function checkAbbreviationAllowedCharacters($abbreviation)
     {
-        $return = false;
-        if (preg_match('/^[a-zA-Z0-9_]*$/', $abbreviation)) {
-            $return = true;
-        }
-        return $return;
+        return preg_match('/^[a-zA-Z0-9_]*$/', $abbreviation);
```

Comment:
Btw. i suggest throwing an Exception in case of Error (even if it's unlikely that error may happen, it can because malformed utf8 or other things):
pseudo code:
```
$res = preg_match('/^[a-zA-Z0-9_]*$/', $abbreviation);
if ($res === false) {
      throw new RegExException(preg_last_error(),'/^[a-zA-Z0-9_]*$/', $abbreviation);
}
return $res;
```
If you do not throw an exception, it might be hard to find the reason if the system does not behave like expected, so it's more a general suggestion.